### PR TITLE
Sets up colcon build & test CI.

### PR DESCRIPTION
**Pull request by francocipollone**
# 🎉 New feature

Related to #3 

## Summary
 - Create package.xml to be identified as colcon package.
  - Sets up GH Action CI for colcon build and test.
    - Install colcon-poetry-ros extension in CI.

## Checklist
- [x] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if it affects the public API)

### CODE REVIEW

File: package.xml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0"?>
+<?xml-model href="http://download.ros.org/schema/package_format2.xsd" schematypens="http://www.w3.org/2001/XMLSchema"?>
+<package format="3">
+  <name>maliput_sim</name>
+  <version>0.1.0</version>
+  <description>Maliput simulator</description>
```
Comment (agalbachicar) [inline package.xml:6]:
```suggestion
  <description>A lightweight simulator for agent development using maliput.</description>
```

Comment (agalbachicar) [reply package.xml:6]: Same as in pyproject.toml

**Review by agalbachicar — APPROVED** [review]
LGTM

File: package.xml (hunk 1 above)
Comment (francocipollone) [reply package.xml]: Done

**Pull request merged**